src/infra/repos/postgresql/guardian-account-repository.ts

```diff
@@ -0,0 +1,71 @@
+
+import { prisma as db } from './prisma'
+import {
+  type LoadGuardianByEmailRepository,
+  type AddGuardianRepository,
+  type UpdateAccessTokenRepository,
+  type LoadGuardianByIdRepository
+} from '@/data/protocols'
+
+export class GuardianAccountRepository implements AddGuardianRepository, LoadGuardianByEmailRepository, LoadGuardianByIdRepository, UpdateAccessTokenRepository {
+  async add (guardianData: AddGuardianRepository.Params): Promise<AddGuardianRepository.Result> {
+    let success: boolean = false
+    const guardianHasEmailRegistered = await db.guardian.findUnique({
+      where: { email: guardianData.email }
+    })
+
+    const guardianHasPhoneRegistered = await db.guardian.findUnique({
+      where: { phone: guardianData.phone }
+    })
+
+    if (!guardianHasEmailRegistered && !guardianHasPhoneRegistered) {
+      await db.guardian.create({
+        data: guardianData
+      })
+
+      success = true
+    }
+    return success
+  }
+
+  async loadByEmail (email: LoadGuardianByEmailRepository.Params): Promise<LoadGuardianByEmailRepository.Result | null> {
+    const guardianDb = await db.guardian.findUnique({ where: { email } })
+    if (!guardianDb) {
+      return null
+    }
+    const guardian = {
+      id: guardianDb.id,
+      firstName: guardianDb.firstName,
+      lastName: guardianDb.lastName,
+      email: guardianDb.email,
+      password: guardianDb.password,
+      phone: guardianDb.phone,
+      accessToken: guardianDb.accessToken,
+      isPrivacyPolicyAccepted: guardianDb.isPrivacyPolicyAccepted
+    }
+    return guardian
+  }
+
+  async loadById (id: LoadGuardianByIdRepository.Params): Promise<LoadGuardianByIdRepository.Result | null> {
+    const guardianDb = await db.guardian.findUnique({ where: { id } })
+    if (!guardianDb) {
+      return null
+    }
+    const guardian = {
```

Comment:
Why do I need to create a variable that is a copy of guardianDb instead of returning guardianDb?

Reply:
Because guardianDb is a instance of prisma, so transform in domain object make sense to don't carry metadata or use prisma types out of infra context.